mii/client.py

```diff
@@ -156,6 +160,38 @@ def destroy_session(self, session_id):
             client.destroy_session(session_id)
 
 
+class MIINonPersistentClient():
+    def __init__(self, task, deployment_name):
+        self.task = task
+        self.deployment_name = deployment_name
+
+    def query(self, request_dict, **query_kwargs):
+        task_methods = GRPC_METHOD_TABLE[self.task]
+        inference_pipeline = mii.non_persistent_models[self.deployment_name][0]
+
+        if self.task == Tasks.QUESTION_ANSWERING:
+            if 'question' not in request_dict or 'context' not in request_dict:
+                raise Exception(
+                    "Question Answering Task requires 'question' and 'context' keys")
+            args = (request_dict["question"], request_dict["context"])
+            kwargs = query_kwargs
+
+        elif self.task == Tasks.CONVERSATIONAL:
+            conv = task_methods.create_conversation(request_dict, **query_kwargs)
+            args = (conv, )
+            kwargs = {}
+
+        else:
+            args = (request_dict['query'], )
+            kwargs = query_kwargs
+
+        return task_methods.run_inference(inference_pipeline, args, query_kwargs)
+
+    def terminate(self):
+        print("Terminating ...")
+        del mii.non_persistent_models[self.deployment_name]
```

Comment:
It would be helpful if we contain the deployment name in the message.

Reply:
I will add this in now.